lighthouse-core/gather/drivers/connection.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const log = require('../../lib/log.js');
+
+class Connection {
+
+  constructor() {
+    this._lastCommandId = 0;
+    /** @type {!Map<number, function(object)}*/
+    this._callbacks = new Map();
+    this._eventEmitter = new EventEmitter();
+  }
+
+  /**
+   * @return {!Promise}
+   */
+  connect() {
+    return Promise.reject(new Error('Not implemented'));
+  }
+
+  /**
+   * @return {!Promise}
+   */
+  disconnect() {
+    return Promise.reject(new Error('Not implemented'));
+  }
+
+  /**
+   * Call protocol methods
+   * @param {!string} method
+   * @param {!Object} params
+   * @return {!Promise}
+   */
+  sendCommand(method, params) {
+    this.formattedLog('method => browser', {method: method, params: params}, 'verbose');
+    var id = ++this._lastCommandId;
+    var message = JSON.stringify({id: id, method: method, params: params || {}});
```

Comment:
> given our node support range {id, method, params} is a no go atm 🐐

we're good on this with node v4: http://node.green/#object-literal-extensions You're maybe thinking of destructuring?